ui/src/main/java/edu/wpi/grip/ui/pipeline/SourceView.java

```diff
@@ -43,9 +49,13 @@ public SourceView(EventBus eventBus, Source source) {
 
         this.name.setText(source.getName());
 
+        controls.getChildren().add(SourceControlsViewFactory.createSourceControlsView(eventBus, source));
```

Comment:
"Source controls" should probably just be subclasses of `SourceView`, not controls inside of it.  See `PreviewView`, which has worked out pretty well for writing new types of previews.